components/brave_new_tab_ui/components/search/config.ts

```diff
@@ -26,38 +22,57 @@ const getConfig = () => {
   return cache!
 }
 
-// Determines whether any search engines are enabled
-export const hasEnabledEngine = () => Object.values(getConfig()).find(c => c)
+export function useNtpWidgetDefaultSearchEngine() {
+  const [lastUsed, setLastUsed] = useNewTabPref('lastUsedNtpSearchEngine')
+  const lastLocalStorage = localStorage.getItem(LAST_SEARCH_ENGINE_KEY)
+  const config = getEnabledSearchEngines()
 
-export const maybeEnableDefaultEngine = () => {
-  // If no engines are enabled, enable the default one.
-  if (!hasEnabledEngine()) {
-    setEngineEnabled({ host: braveSearchHost }, true)
-  }
-}
+  const setLastNtpSearchEngine = React.useCallback((engine: SearchEngineInfo) => {
+    setLastUsed(engine.host);
+    localStorage.setItem(LAST_SEARCH_ENGINE_KEY, engine.host);
+  }, [setLastUsed])
 
-export const setEngineEnabled = (engine: { host: string }, enabled: boolean) => {
-  const config = getConfig()
-  config[engine.host] = enabled
+  const lastNtpSearchEngine = React.useCallback(() => {
+    let last = lastUsed;
 
-  localStorage.setItem(ENABLED_SEARCH_ENGINES_KEY, JSON.stringify(config))
-}
+    if (lastLocalStorage && config[lastLocalStorage] && lastUsed !== lastLocalStorage) {
+      setLastUsed(lastLocalStorage);
+      last = lastLocalStorage;
+    }
 
-export const isSearchEngineEnabled = (engine: SearchEngineInfo) => getConfig()[engine.host]
+    // If the last search engine we used has been disabled or doesn't exist in the config, return the first enabled
+    // one, or Brave Search.
+    if (!last || !config[last]) {
+      return Object.keys(config).find(key => config[key]) ?? braveSearchHost
+    }
 
-export const getDefaultSearchEngine = () => {
-  const config = getConfig()
-  const last = localStorage.getItem(LAST_SEARCH_ENGINE_KEY)
+    return last
+  }, [lastUsed, lastLocalStorage, config]);
 
-  // If the last search engine we used has been disabled, return the first enabled
-  // one, or Brave Search.
-  if (!config[last!]) {
-    return Object.entries(config).find(([key, value]) => value)?.[0]
-      ?? braveSearchHost
+  return {
+    lastNtpSearchEngine: lastNtpSearchEngine(),
+    setLastNtpSearchEngine
   }
-  return last
 }
 
-export const setDefaultSearchEngine = (engine: SearchEngineInfo) => {
-  localStorage.setItem(LAST_SEARCH_ENGINE_KEY, engine.host)
+export function useEnabledSearchEngines() {
+  const config = getEnabledSearchEngines()
+
+  const setEngineEnabled = React.useCallback((engine: { host: string }, enabled: boolean) => {
+    config[engine.host] = enabled
```

Comment:
Hey so calling this function will **not** necessarily trigger a rerender. You need to have something in state which gets changed. The easiest way is probably to mirror the change in state, or have a flag variable you set.

```ts
const [,setRenderCount] = useState(0)

// in setEngineEnabled:
setRenderCount(r => r + 1);
```

or 

```ts
const [config, setConfig] = useState(getEnabledSearchEngines())

// in setEngineEnabled:
// Note: This needs to be a difference object, hence the spread
setConfig({...getEnabledSearchEngines})
```